TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java

```diff
@@ -183,8 +183,12 @@ private void writeTimestampForNowToCurrentStream(Instant timestamp) throws IOExc
     }
 
     private void writeByteBufferToCurrentStream(int fieldNum, ByteBuffer byteBuffer) throws IOException {
-        if (byteBuffer.remaining() > 0) {
-            getOrCreateCodedOutputStream().writeByteBuffer(fieldNum, byteBuffer);
+        if (byteBuffer.hasRemaining()) {
+            // CodedOutputStream.writeByteBuffer writes based on capacity and ignores limits so prefer write
+            getOrCreateCodedOutputStream().writeTag(fieldNum, WireFormat.WIRETYPE_LENGTH_DELIMITED);
```

Comment:
The comment above
> // CodedOutputStream.writeByteBuffer writes based on capacity and ignores limits so prefer write

Reply:
The alternative is to use CodedOutputStream.writeByteBuffer(fieldNum, byteBuffer.slice()), though i was trying to avoid using slice to have greater clarity.

I'm also going to change this method to modify the position of the passed in byteBuffer so to achieve that we would have had to do

```
var slicedByteBuf = byteBuffer.slice();
getOrCreateCodedOutputStream().writeByteBuffer(fieldNum, slicedByteBuf);
byteBuffer.position(slicedByteBuf.position());
```

let me know if you would prefer that to
```
var codedOutputSream = getOrCreateCodedOutputStream();
codedOutputSream.writeTag(fieldNum, WireFormat.WIRETYPE_LENGTH_DELIMITED);
codedOutputSream.writeUInt32NoTag(byteBuffer.remaining());
codedOutputSream.write(byteBuffer);
```

Reply:
Fixed this with a further refactor that cleaned up and consolidated the interface between Netty ByteBufs and nio ByteBuffers